OSDOCS-9641: Scaled registry NFS docs (#78468)
openshift/openshift-docs: modules/recommended-configurable-storage-technology.adoc

* Object storage should be S3 or Swift compliant.
* For non-cloud platforms, such as vSphere and bare metal installations, the only configurable technology is file storage.
* Block storage is not configurable.

> * Using Network File System (NFS) storage with {product-title} is supported. However, using NFS storage with a scaled registry can cause known issues and is not recommended. For more information, see the Red{nbsp}Hat Knowledgebase solution link:https://access.redhat.com/solutions/3428661[Is NFS supported for OpenShift cluster internal components in Production?].

Reviewer: ```suggestion
* The use of Network File System (NFS) storage with {product-title} is supported. However, the use of NFS storage with a scaled registry can cause known issues. For more information, see the Red{nbsp}Hat Knowledgebase solution, link:https://access.redhat.com/solutions/3428661[Is NFS supported for OpenShift cluster internal components in Production?].
```
- Revised "using" to "the use of" to avoid ambiguity, per IBM Style. See _using_ on the [Word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage) page.
- Removed "not recommended", as that phrase can be tricky for legal reasons. I think it's sufficient to state that the use of NFS storage with a scaled registry can cause known issues.

Author: Great comments, thanks Laura!
